avocado/utils/configure_network.py

```diff
@@ -35,168 +37,248 @@ class NWException(Exception):
+            return genio.read_file("/sys/class/net/%s/operstate" % self.name)
 
-def set_mtu_host(interface, mtu):
-    """
-    Set MTU size in host interface
-    """
-    cmd = "ip link set %s mtu %s" % (interface, mtu)
-    try:
-        process.system(cmd, shell=True)
-    except process.CmdError as ex:
-        raise NWException("MTU size can not be set: %s" % ex)
-    try:
-        cmd = "ip add show %s" % interface
-        mtuvalue = process.system_output(cmd, shell=True).decode("utf-8") \
-                                                         .split()[4]
-        if mtuvalue == mtu:
+    def is_link_up(self):
+        """
+        Checks if the interface link is up
+        :return: True if the interface's link up, False otherwise.
+        """
+        if self.get_link_state() in ['up', 'UP']:
             return True
-    except Exception as ex:  # pylint: disable=W0703
-        log.error("setting MTU value in host failed: %s", ex)
-    return False
+        else:
+            return False
 
+    def bring_up(self):
+        """
+        utility used to Bring up interface
+        """
+        cmd = "ifup %s" % self.name
+        try:
+            process.system(cmd, ignore_status=False, sudo=True)
+            return True
+        except process.CmdError as ex:
+            raise NWException("ifup fails: %s" % ex)
 
-class PeerInfo:
-    """
-    class for peer function
-    """
+    def bring_down(self):
+        """
+        Utility  Used to Bring down interface
+        """
+        cmd = "ifdown %s" % self.name
+        try:
+            process.system(cmd, sudo=True)
+            return True
+        except Exception as ex:
+            raise NWException("ifdown fails: %s" % ex)
 
-    def __init__(self, host, port=None, peer_user=None,
-                 key=None, peer_password=None):
+    def _move_config_file(self, src_conf, dest_conf):
+        if os.path.exists(src_conf):
+            shutil.move(src_conf, dest_conf)
+        else:
+            raise NWException("%s interface not available" % self.name)
+
+    def get_hwaddr(self):
+        try:
+            interface_file = open(
+                '/sys/class/net/%s/address' % self.name)
+            hwaddr = interface_file.read().strip()
+            interface_file.close()
+            return hwaddr
+        except OSError as ex:
+            raise NWException("interface not found : %s" % ex)
+
+    def set_hwaddr(self, hwaddr):
         """
-        create a object for accesses remote machine
+        Utility which set Hw address to Interface
+        :param hwaddr: Pass Hardwae address for defined interface
         """
         try:
-            self.session = Session(host, port=port, user=peer_user,
-                                   key=key, password=peer_password)
-        except Exception as ex:  # pylint: disable=W0703
-            log.error("connection not established to peer machine: %s", ex)
+            process.system('ifconfig %s hw ether %s' %
```

Comment:
Sometimes you use `ifconfig`, sometimes `ip`. Is there any reason to not use `iproute` here?

```bash
$  ip link set eth0 address 02:01:02:03:04:08
```